surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,331 @@
+- Visual Studio Code for `gdb`
+- Visual Studio Code for `lldb`
+- `gdb`'s TUI mode
+- Vim/Neovim's `termdebug` for `gdb`
+- Pernosco for `rr`
+- ...
+
+> *Justification:*
+>
+> Interface used for a debugger materially changes the experience of using a
+> debugger and the advantages/disadvantages of it.
+
+### What pain points have you experienced with a debugger?
+Type: select all that apply
+
+- Often step too many times and need to restart from scratch
+- Don't have a source view of the current line being run
+- Unintuitive interface/errors
+- Expression parser doesn't support expressions that you want to write
+- Optimized-out code
+- Conditional breakpoints don't trigger
+- Difficult to trace dataflow
+- Too much detail/irrelevant information (i.e. assembly views)
+- Incorrect information displayed (i.e. line numbers, variable values)
+
+> *Justification:*
+>
+> It might be possible for us to address these - or they might be the result of
+> the compiler's debuginfo output.
+
+
+### Does your debugging typically cross language boundaries?
+Type: yes/no
+
+> *Justification:*
+>
+> If lots of users are using debuggers when doing cross-language interop, then
+> support for this could be improved.
+
+### Which programming language(s) do you debug across?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- ...
+
+> *Justification:*
+>
+> Debugging Rust + C is very different from Rust + Python and the quality of
+> the experience is likely very different too.
+
+
+### Have you ever written a visualization for a debugger?
+Type: select all that apply
+
+- Yes, for `gdb`
+- Yes, for `WinDbg`
+- ...
+
+> *Justification:*
+>
+> Part of high-quality debugger support is things like type visualizations that
+> could be shipped by libraries, knowing what experience Rust users have with
+> writing these visualizations would be valuable.
+
+
+### Have you ever used a debugger on macro-generated code?
+Type: yes/no
+
+> *Justification:*
+>
+> Macros are frequently used in Rust and has unique challenges for debugging.
+
+### What are your experiences using a debugger on macro-generated code?
+Type: free-form text
+
+> *Justification:*
+>
+> Macros are frequently used in Rust and has unique challenges for debugging.
+
+### Have you ever used a debugger on a procedural macro?
+Type: yes/no
+
+> *Justification:*
+>
+> Macros are frequently used in Rust and has unique challenges for debugging.
+
+### What are your experiences using a debugger on a procedural macro?
+Type: free-form text
+
+> *Justification:*
+>
+> Macros are frequently used in Rust and has unique challenges for debugging.
+
+### Have you ever used a debugger with async Rust?
+Type: yes/no
+
+> *Justification:*
+>
+> async Rust is frequently used and has unique challenges for debugging.
+
+### What are your experiences using a debugger on async Rust?
+Type: free-form text
+
+> *Justification:*
+>
+> async Rust is frequently used and has unique challenges for debugging.
```

Comment:
Some variation on this question might also be interesting: When debugging how often do you need to debug into code that is not your own (either std lib or crates.io)?

Reply:
Added!